### ISSUE TITLE
And after Then should not rerun setup

### ISSUE THREAD

**Issue by tandrewnichols** [rendro/mocha-given#2]
I use jasmine-given at work (and have even worked with Justin Searls on writing some parts of it), and one of the fundamental goals is to only run setup (`Given`, `When`) once per `Then`. I've noticed in my tests using mocha-given that

``` coffee
Then -> testSomething()
And -> testSomethingElse()
```

will run all `Given`s and `When`s twice. Which kind of defeats the purpose. Now, to only run setup once, instead of getting the nice formatting of cascading `And`s, I have to put everything in one giant `Then` block. If you check out https://github.com/searls/jasmine-given/blob/master/app/js/jasmine-given.coffee#L50 and https://github.com/searls/jasmine-given/blob/master/app/js/jasmine-given.coffee#L82, the `subsequentThen` mechanism tacks `And` functions on so that they're executed as part of the same `it` statement.

In other words I want to do something like:

``` coffee
Given -> @foo = 'bar'
When -> @subject.test()
Then -> expect(@foo).to.equal 'not bar'
And -> expect(@subject.aSpiedOnMethod).to.have.been.called
And -> expect(@someOtherStuff).to.have.happened()
```

and have the `Given` and `When` run once instead of three times.

I'm willing to work on this (or work on this with you) if you want.
